### DISCUSSION TITLE
How to find record in table by hex value in bytea column?

### DISCUSSION THREAD

**Discussion by stasvo** [jackc/pgx#1649] [Q&A] [answered]
I have a problem of find records in tablbe by hex string in bytea column when use argument.
For example I have hash 83d0c8c41e14f33c94cdb5a9b875c418e82ebd82 which stored in bytea column.
If I use query without arguments, in this way, then its works without problems: 
```select bytea_column from table where bytea_column = '\x83d0c8c41e14f33c94cdb5a9b875c418e82ebd82';``` 
But is I use it as strign argument it doesnt works in my case. 
```select bytea_column from table where bytea_column = $1;``` 
I tried pass '\x83d0c8c41e14f33c94cdb5a9b875c418e82ebd82' to argument as string.
I tried pass '83d0c8c41e14f33c94cdb5a9b875c418e82ebd82' to argument as string.
I tried this way which also didnt work:
```
decoded, err := hex.DecodeString("83d0c8c41e14f33c94cdb5a9b875c418e82ebd82")
$1 = decoded
```
What a right way to use this value as argument for bytea?

**Comment by jackc** [comment, marked as answer]
You should be able to use a `[]byte` as an `bytea` parameter. `hex.DecodeString` should have done what you want.

```go
package main

import (
	"context"
	"encoding/hex"
	"log"
	"os"

	"github.com/jackc/pgx/v5"
)

func main() {
	ctx := context.Background()

	conn, err := pgx.Connect(ctx, os.Getenv("DATABASE_URL"))
	if err != nil {
		log.Fatal(err)
	}
	defer conn.Close(ctx)

	b, err := hex.DecodeString("01020304")
	if err != nil {
		log.Fatal(err)
	}

	var isEqual bool
	err = conn.QueryRow(ctx, `select '\x01020304'::bytea = $1::bytea`, b).Scan(&isEqual)
	if err != nil {
		log.Fatal(err)
	}

	log.Println("isEqual:", isEqual)
}
``` 

```
jack@glados ~/dev/pgx_issues/pgx-1649 ±master⚡ » go run .
2023/06/18 06:56:25 isEqual: true
```

**Reply by stasvo** [reply]
Thank you.
